src/category_theory/sites/sieves.lean

```diff
@@ -219,6 +219,31 @@ def gi_generate : galois_insertion (generate : presieve X → sieve X) arrows :=
   choice_eq := λ _ _, rfl,
   le_l_u := λ S Y f hf, ⟨_, 𝟙 _, _, hf, category.id_comp _⟩ }
 
+lemma le_generate (R : presieve X) : R ≤ generate R :=
+gi_generate.gc.le_u_l R
+
+/-- If the identity arrow is in a sieve, the sieve is maximal. -/
+lemma id_mem_iff_eq_top : S (𝟙 X) ↔ S = ⊤ :=
+⟨λ h, top_unique $ λ Y f _, by simpa using downward_closed _ h f,
+ λ h, h.symm ▸ trivial⟩
+
+/-- If an arrow set contains a split epi, it generates the maximal sieve. -/
+lemma generate_of_contains_split_epi {R : presieve X} (f : Y ⟶ X) [split_epi f]
+  (hf : R f) : generate R = ⊤ :=
+begin
+  rw ← id_mem_iff_eq_top,
+  exact ⟨_, section_ f, f, hf, by simp⟩,
+end
+
+@[simp]
+lemma generate_of_singleton_split_epi (f : Y ⟶ X) [split_epi f] :
```

Comment:
Should I expect `simp` to be able to use this? I don't mind removing the simp attribute

Reply:
Yes, I think this is fine.